Fixed #26993 -- Increased User.last_name max_length to 150 characters.

thomwiggers opened the pull request:
Increases the max_length of last_name in django.contrib.auth to 100 characters to facilitate longer last_names.

Fixes [ticket 26993](https://code.djangoproject.com/ticket/26993)

timgraham commented on docs/ref/contrib/auth.txt:
```diff
@@ -47,7 +47,11 @@ Fields
+
+        .. versionchanged:: 2.0
+
+            Increased from a maximum of 30 characters to a maximum of 100 characters.
```
Please follow the wording in 15ef1dd478c5b6f005e5f782b7619f718ed55e84.

timgraham commented on docs/releases/2.0.txt:
```diff
@@ -54,6 +54,8 @@ Minor features
 * The default iteration count for the PBKDF2 password hasher is increased from
   36,000 to 100,000.
 
+* The maximum length of `User.last_name` is increased to 100 characters.
```
This wasn't considered a "feature" in 15ef1dd478c5b6f005e5f782b7619f718ed55e84 so I think we can also omit it here.

thomwiggers replied:
I thought it similar to the hashing round increase but omitting it is fine.

timgraham commented on docs/releases/2.0.txt:
```diff
@@ -210,6 +212,36 @@ Database backend API
+
+    from django.contrib.auth.forms import UserCreationForm
+
+    class MyUserCreationForm(UserCreationForm):
```
Some thinking is needed here. Unlike the `username` field, the `UserCreationForm` doesn't have `last_name` field. This needs to be reflected on the user change form instead.

thomwiggers commented:
Changes:
-------------
* Rebased
* Now 150 characters (fine by me: my 'long name' benchmark, the Dutch royal family, maxed out at ~110)
* Changed wording as suggested
* Now the changelog suggests to edit `UserChangeForm` if wanted.
* I removed the `required` from the `last_name` field's `help_text` in the example form, as it isn't necessarily true for `last_name` as it was for `username`.
